### ISSUE TITLE
Problem with aws_launch_configuration_invalid_image_id

### ISSUE THREAD

**Issue by jhedev** [wata727/tflint#447]
tflint is reporting "invalid image ID" on a launch configuration in our codebase. However, there are instances running with this AMI so it's obviously not an issue. 

The AMI is create in a different account and shared with this account. Could this cause an issue with the validation using `DescribeImages`?

**Comment by wata727** [comment]
Umm, this is a difficult problem...

Unfortunately, I don’t have an account with shared AMI, so it cannot be reproduced in my environment. But that is probably the cause. The place where `DescribeImages` is invoked is as follows:
https://github.com/wata727/tflint/blob/08afbfa51ed6cbd4bf6d41c43de8910515bb098b/rules/awsrules/aws_launch_configuration_invalid_image_id.go#L61-L63

On the other hand, according to the description of aws-cli, it says "private images owned by other AWS accounts for which you have explicit launch permissions are available":
https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-images.html#description

It’s difficult to investigate this issue, so it’s likely that it will take time. If you would like to ignore these false positives, you can use `tflint-ignore` annotation comments. See https://github.com/wata727/tflint/blob/v0.11.2/docs/guides/config.md#rule-blocks

**Comment by jhedev** [comment]
Thanks for your answer. I suspect the same line. Maybe I'll find some time to look into it myself.

**Comment by jhedev** [comment]
Interestingly, if I use `aws  ec2 describe-images --image-ids <image-id>` with the same profile it lists the correct image...

**Comment by jhedev** [comment]
Alright, it seems to be a credentials issue. If I set up the right profile on my laptop it works fine now. In the terraform setup we use role assumption to log in to the other account. At first glance it looked like tflint takes this information from the tf provider setup, is this correct?

**Comment by jhedev** [comment]
I found the cause of the issue we are seeing. Currently, we use the account ID as a variable in our `assume_role` block in the provider config, like this:

```hcl
  assume_role {
    role_arn = "arn:aws:iam::${var.account_id}:role/terraform"
  }
```

If I change the variable to the real account ID it works fine. So I guess interpolation in the provider config is not supported by tflint?

**Comment by wata727** [comment]
Thank you for your investigation. Certainly assume_role is supported by this pull request https://github.com/wata727/tflint/pull/423

However, interpolation is supported in the provider block. Some error may have occurred. Could you run it with `TFLINT_LOG=debug`? You can get a lot of output, but the first part is important in the case.

**Comment by jhedev** [comment]
Looks like I got it working now... The problem was the user :D I did not correctly set up the var files with tflint. So it simply didn't know all the variable values...

Thanks for your help and sorry for bothering you :)
